src/extension/public/popup.js

```diff
@@ -1,58 +1,86 @@
-const generateToken= document.querySelector('.generate-cli-token');
-async function getCurrentTab() {
-    const queryOptions = { active: true, lastFocusedWindow: true };
-    const [tab] = await chrome.tabs.query(queryOptions);
-    return tab;
-}
-document.addEventListener('DOMContentLoaded',async ()=>{
-    const currentTab = await getCurrentTab();
-    if(currentTab.url.indexOf('gitlab') > -1){
-        generateToken.style.display='block';
-    }
-});
-generateToken.addEventListener('click',async ()=>{
-    const currentTab = await getCurrentTab();
-    const currentURL = new URL(currentTab.url);
-    const ORIGN = currentURL.origin;
-    chrome.storage.sync.set({ 'gitTokenCopy':true });
-    chrome.tabs.create({active: true, url: `${ORIGN}/-/profile/personal_access_tokens?create_gitlabcli_token=true`});
-})
-
-function renderTokenValue(){
-    const privateTokenInput = form.elements['private-token'];
-    try{
-        chrome.storage.sync.get(['gitToken'], function(items) {
-            if(!(items.gitToken) || (items.gitToken).trim() === ''){
-                privateTokenInput.value = 'No Tokens Found!'
+function streamBody(body, onChunkReceive) {
```

Comment:
why did we move these functions in popup.js ?

Reply:
Previously fetchstream only made Get requests. Modified it for post request now.
Changes done!